ui-charts/src/manchette/hooks/useManchetteWithSpaceTimeChart.ts

```diff
@@ -362,161 +334,361 @@ const useManchetteWithSpaceTimeChart = ({
+      } else {
+        allScales.push({
+          to: baseScale.to,
+          coefficient,
+        });
+      }
+    });
+
+    splitPoints.forEach((point) => {
+      allScales.push({
+        to: point.position,
+        size: point.size,
+      });
+    });
+
+    return sortBy(allScales, (scale) => scale.to + ('size' in scale ? Number.EPSILON : 0));
+  }, [
+    waypoints,
+    height,
+    isProportional,
+    yZoom,
+    minZoomMillimeterPerPx,
+    maxZoomMillimeterPerPx,
+    splitPoints,
+  ]);
+
+  const waypointsWithoutSplitPoints = useMemo(() => {
+    const splitPointPositions = new Set(splitPoints?.map((point) => point.position) || []);
+    const filteredWaypoints = selectWaypointsToDisplay(waypoints, {
+      height,
+      isProportional,
+      yZoom,
+    });
+    return filteredWaypoints.filter((waypoint) => !splitPointPositions.has(waypoint.position));
+  }, [splitPoints, waypoints, height, isProportional, yZoom]);
+
+  const { manchetteContents, manchetteHeight } = useMemo(() => {
+    const spaceScaleTree = spaceScalesToBinaryTree(spaceOrigin, spaceScales);
+    const getSpacePixel = getSpaceToPixel(0, spaceScaleTree);
+    const totalManchetteHeight = getSpacePixel(spaceScales.at(-1)!.to, true) + BASE_WAYPOINT_HEIGHT;
+
+    if (!splitPoints)
+      return {
+        manchetteHeight: totalManchetteHeight,
+        manchetteContents: waypointsWithoutSplitPoints,
+      };
+
+    // Identify all manchette contents (split sections and waypoints):
+    let allContents: (
+      | { type: 'waypoint'; position: number; waypoint: Waypoint }
+      | { type: 'splitSection'; position: number; split: SplitPoint }
+    )[] = waypointsWithoutSplitPoints.map((wp) => ({
+      type: 'waypoint',
+      waypoint: wp,
+      position: wp.position,
+    }));
+    allContents = allContents.concat(
+      splitPoints.map((sp) => ({
+        type: 'splitSection',
+        split: sp,
+        position: sp.position,
+      }))
+    );
+
+    // Sort all contents by position:
+    const allSortedContents = sortBy(allContents, 'position');
+
+    // Iterate over all contents, to set each split section's style, and correct waypoints styles
+    // accordingly:
+    const finalContents: (InteractiveWaypoint | ReactNode)[] = [];
+    allSortedContents.forEach((content) => {
+      switch (content.type) {
+        case 'waypoint': {
+          finalContents.push({
+            ...content.waypoint,
+            styles: {
+              top: `${getSpacePixel(content.position)}px`,
+              height: `${BASE_WAYPOINT_HEIGHT}px`,
+            },
+          });
+          break;
+        }
+        case 'splitSection': {
+          const {
+            split: { position, size, payload },
+          } = content;
+
+          const top = getSpacePixel(position);
+          finalContents.push(
+            createElement(
+              'div',
+              {
+                style: {
+                  top: `${top + WAYPOINTS_OFFSET}px`,
```

Comment:
Why do we need to apply an offset here, but not for "regular" waypoints?

Reply:
The reason is:
- Waypoints are positioned so that the line at "the middle" is aligned with the space pixel
- Split points are positioned so that their top is aligned with the space pixel

Reply:
Ah right, thanks for the explanation!